iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformers.java

```diff
@@ -0,0 +1,490 @@
+        return isWellOriented ? t2w.getOperationalLimitsGroups1() : t2w.getOperationalLimitsGroups2();
+    }
+
+    private static Terminal getTerminal1(TwoWindingsTransformer t2w, boolean isWellOriented) {
+        return isWellOriented ? t2w.getTerminal1() : t2w.getTerminal2();
+    }
+
+    private static Terminal getTerminal2(TwoWindingsTransformer t2w, boolean isWellOriented) {
+        return isWellOriented ? t2w.getTerminal2() : t2w.getTerminal1();
+    }
+
+    private static String getEnd1(boolean isWellOriented) {
+        return isWellOriented ? "1" : "2";
+    }
+
+    private static Map<Terminal, Terminal> getRegulatedTerminalMapping(TwoR twoR, ThreeWindingsTransformer t3w) {
+        Map<Terminal, Terminal> regulatedTerminalMapping = new HashMap<>();
+        regulatedTerminalMapping.put(getTerminal1(twoR.t2w1, isWellOriented(twoR.starBus, twoR.t2w1)), t3w.getLeg1().getTerminal());
+        regulatedTerminalMapping.put(getTerminal1(twoR.t2w2, isWellOriented(twoR.starBus, twoR.t2w2)), t3w.getLeg2().getTerminal());
+        regulatedTerminalMapping.put(getTerminal1(twoR.t2w3, isWellOriented(twoR.starBus, twoR.t2w3)), t3w.getLeg3().getTerminal());
+        return regulatedTerminalMapping;
+    }
+
+    private static void replaceRegulatedTerminal(ThreeWindingsTransformer.Leg t3wLeg, Map<Terminal, Terminal> regulatedTerminalMapping) {
+        t3wLeg.getOptionalRatioTapChanger().ifPresent(rtc -> {
+            if (regulatedTerminalMapping.containsKey(rtc.getRegulationTerminal())) {
+                rtc.setRegulationTerminal(regulatedTerminalMapping.get(rtc.getRegulationTerminal()));
+            }
+        });
+        t3wLeg.getOptionalPhaseTapChanger().ifPresent(ptc -> {
+            if (regulatedTerminalMapping.containsKey(ptc.getRegulationTerminal())) {
+                ptc.setRegulationTerminal(regulatedTerminalMapping.get(ptc.getRegulationTerminal()));
+            }
+        });
+    }
+
+    private static void copyStarBusVoltageAndAngle(Bus starBus, ThreeWindingsTransformer t3w) {
+        if (Double.isFinite(starBus.getV()) && starBus.getV() > 0.0 && Double.isFinite(starBus.getAngle())) {
+            t3w.setProperty("v", String.valueOf(starBus.getV()));
+            t3w.setProperty("angle", String.valueOf(starBus.getAngle()));
+        }
+    }
+
+    private static List<PropertyR> copyProperties(TwoWindingsTransformer t2w, ThreeWindingsTransformer t3w) {
+        List<PropertyR> lostProperties = new ArrayList<>();
+        t2w.getPropertyNames().forEach(propertyName -> {
+            boolean copied = copyProperty(propertyName, t2w.getProperty(propertyName), t3w);
+            if (!copied) {
+                lostProperties.add(new PropertyR(t2w.getId(), propertyName));
+            }
+        });
+        return lostProperties;
+    }
+
+    private static boolean copyProperty(String propertyName, String property, ThreeWindingsTransformer t3w) {
```

Comment:
Properties have a strong dependency on the propertyName.  The propertyName is essential to use properly the property data. I'm not sure if copying properties with a general procedure, without adjusting their propertyNames will be the best option as the result will not be in general usable for exporting the model to the original format, the main use of the properties now, I think.

Reply:
Properties are used to pass additional data on an identifiable. They may be used in a full IIDM context (without exporting the network in another format) and it can make sense to transfer them.

Reply:
Done.